Tests/AmplitudeTests/Migration/StoragePrefixMigrationTests.swift

```diff
@@ -0,0 +1,132 @@
+        XCTAssertEqual(destinationLastEventId, 12345)
+        XCTAssertEqual(destinationEventsFileKey as? Int, 789)
+    }
+
+    func testEventFiles() throws {
+        let source = PersistentStorage(storagePrefix: NSUUID().uuidString)
+        let destination = PersistentStorage(storagePrefix: NSUUID().uuidString)
+
+        try source.write(key: StorageKey.EVENTS, value: BaseEvent(eventType: "event-1"))
+        try source.write(key: StorageKey.EVENTS, value: BaseEvent(eventType: "event-2"))
+        source.rollover()
+        try source.write(key: StorageKey.EVENTS, value: BaseEvent(eventType: "event-3"))
+        source.rollover()
+        try source.write(key: StorageKey.EVENTS, value: BaseEvent(eventType: "event-4"))
+
+        var sourceEventFiles = source.getEventFiles(includeUnfinished: true)
+        XCTAssertEqual(sourceEventFiles.count, 3)
+
+        let sourceFileSizes = try sourceEventFiles.map{ try getFileSize($0) }
+
+        var destinationEventFiles = destination.getEventFiles(includeUnfinished: true)
+        XCTAssertEqual(destinationEventFiles.count, 0)
+
+        let migration = StoragePrefixMigration(source: source, destination: destination)
+        migration.execute()
+
+        sourceEventFiles = source.getEventFiles(includeUnfinished: true)
+        XCTAssertEqual(sourceEventFiles.count, 0)
+
+        destinationEventFiles = destination.getEventFiles(includeUnfinished: true)
+        XCTAssertEqual(destinationEventFiles.count, 3)
+
+        for (index, destinationEventFile) in destinationEventFiles.enumerated() {
+            let fileSize = try getFileSize(destinationEventFile)
+            XCTAssertEqual(fileSize, sourceFileSizes[index])
+        }
+    }
+
+    func testMissingSource() throws {
+        let source = PersistentStorage(storagePrefix: NSUUID().uuidString)
+        let destination = PersistentStorage(storagePrefix: NSUUID().uuidString)
+
+        var destinationDeviceId: String? = destination.read(key: StorageKey.DEVICE_ID)
+        var destinationLastEventId: Int64? = destination.read(key: StorageKey.LAST_EVENT_ID)
+        XCTAssertNil(destinationDeviceId)
+        XCTAssertNil(destinationLastEventId)
+
+        let migration = StoragePrefixMigration(source: source, destination: destination)
+        migration.execute()
+
+        destinationDeviceId = destination.read(key: StorageKey.DEVICE_ID)
+        destinationLastEventId = destination.read(key: StorageKey.LAST_EVENT_ID)
+        XCTAssertNil(destinationDeviceId)
+        XCTAssertNil(destinationLastEventId)
+
+        let destinationEventFiles = destination.getEventFiles(includeUnfinished: true)
+        XCTAssertEqual(destinationEventFiles.count, 0)
+
+        let sourceEventsStorageDirectory = source.getEventsStorageDirectory(createDirectory: false)
+        XCTAssertFalse(FileManager.default.fileExists(atPath: sourceEventsStorageDirectory.path))
+    }
+
+    func testDoNotMoveEventFilesToDestinationWithWrittenEvents() throws {
```

Comment:
Do we have to drop these?

The user/session information is persisted in the event storage. So the only difference I can think of would be a change in API key, which is probably ok.

Reply:
The test is removed (see a comment above), new test with duplicated file names is added.